tools/nuget/validate_nuget.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import sys
+import os
+import zipfile # Available Python 3.2 or higher
+import glob
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="ONNX Runtime create nuget spec script",
+                                     usage='')
+    # Main arguments
+    parser.add_argument("--nuget_path", required=True, help="Path containing the Nuget to be validated. Must only contain only one Nuget within this.")
+    parser.add_argument("--platforms_supported", required=True, help="Comma separated list (no space). Ex: linux-x64,win-x86,osx-x64")
+    parser.add_argument("--verify_nuget_signing", required=True, help="Flag inidicating if Nuget package signing is to be verified. Only accepets 'true' or 'false'")
+    
+    return parser.parse_args()
+
+def check_exists(path):
+    return os.path.exists(path)
+   
+def is_windows():
+    return sys.platform.startswith("win")
+   
+def check_if_dlls_are_present(platforms_supported, zip_file):
+    platforms = platforms_supported.strip().split(",")
+    for platform in platforms:
+        if platform.startswith("win"):
+            path = "runtimes/" + platform + "/native/onnxruntime.dll"
+            print('Checking path: ' + path)
+            if (not path in zip_file.namelist()):
+               print("onnxruntime.dll not found for " + platform)
+               print(zip_file.namelist())
+               raise Exception("onnxruntime.dll not found for " + platform)
+
+        elif platform.startswith("linux"):
+            path = "runtimes/" + platform + "/native/libonnxruntime.so"
+            print('Checking path: ' + path)
+            if (not path in zip_file.namelist()):
+               print("libonnxruntime.so not found for " + platform)
+               raise Exception("libonnxruntime.so not found for " + platform)
+               
+        elif platform.startswith("osx"):
+            path = "runtimes/" + platform + "/native/libonnxruntime.dylib"
+            print('Checking path: ' + path)        
+            if (not path in zip_file.namelist()):
+               print("libonnxruntime.dylib not found for " + platform)
+               raise Exception("libonnxruntime.dylib not found for " + platform)
+               
+        else:
+            raise Exception("Unsupported platform: " + platform)
+            
+def check_if_nuget_is_signed(nuget_path):
```

Comment:
This step might be confusing as we have a standalone step in the pipeline that does the signing and if that step fails and the pipeline stops, this verification seems redundant. But during one of the releases, we had an accidental scenario wherein the nuget that was composed in the BinariesDirectory was copied over to the ArtifactStagingDirectory and then the nuget in the BinariesDirectory was signed successfully (but this is of no use as we rely on the one in the ArtifactStagingDirectory). This specific logic added makes sure the Nuget in the ArtifactStagingDirectory is signed.